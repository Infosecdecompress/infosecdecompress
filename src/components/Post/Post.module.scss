@import "../../assets/scss/variables";
@import "../../assets/scss/mixins";

.post {
  .content {
    @include margin-auto();
  }

  .footer {
    max-width: $layout-post-width;
    @include padding(0, 0.5);
    @include margin-auto();
  }

  .comments {
    max-width: $layout-post-width;
    @include padding(0, 0.5);
    @include margin-auto();
  }

  .button {
    border: 1px solid $color-gray-border;
    border-radius: $button-border-radius;
    color: $color-base;
    display: block;
    font-size: $typographic-base-font-size;
    font-weight: normal;
    height: $button-height;
    line-height: $button-height;
    margin-left: auto;
    margin-right: auto;
    max-width: 90px;
    text-align: center;
    @include padding(0, 1);
    @include margin-top(1);

    &:hover,
    &:focus {
      color: $color-primary;
    }
  }
}

@include breakpoint-md {
  .post {
    .footer {
      @include padding-equal(0);
    }

    .comments {
      @include padding-equal(0);
    }

    .button {
      left: 30px;
      max-width: none;
      position: fixed;
      top: 30px;
      @include margin-equal(0);
    }
  }
}
