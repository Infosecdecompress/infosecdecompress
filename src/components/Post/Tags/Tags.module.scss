@import "../../../assets/scss/variables";
@import "../../../assets/scss/mixins";

.tags {
  @include margin-bottom(0.5);

  .list {
    list-style: none;
    @include padding-equal(0);

    .item {
      display: inline-block;
      @include margin(0.5, 0.125);

      @include breakpoint-sm {
        &:first-child {
          @include margin-left(0);
          @include padding-left(0);
        }
      }

      .link {
        border: 1px solid $color-gray-border;
        border-radius: $button-border-radius;
        color: $color-base;
        display: inline-block;
        height: $button-height;
        line-height: $button-height;
        text-decoration: none;
        @include padding(0, 0.75);

        &:hover,
        &:focus {
          color: $color-primary;
        }
      }
    }
  }
}
