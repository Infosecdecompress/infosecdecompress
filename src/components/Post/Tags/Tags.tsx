import React from "react";

import { Link } from "gatsby";

import * as styles from "./Tags.module.scss";

type Props = {
  tags: string[];
  tagSlugs: string[];
};

const Tags = ({ tags, tagSlugs }: Props) => (
  <div className={styles.tags}>
    <ul className={styles.list}>
      {tagSlugs
        ? tagSlugs.map((slug, i) => (
            <li className={styles.item} key={slug}>
              <Link to={slug} className={styles.link}>
                {tags[i]}
              </Link>
            </li>
          ))
        : null}
    </ul>
  </div>
);

export default Tags;
