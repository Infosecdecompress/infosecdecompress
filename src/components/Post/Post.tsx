import React from "react";

import { Link } from "gatsby";

import type { Node } from "@/types";

import { Author } from "./Author";
import { Comments } from "./Comments";
import { Content } from "./Content";
import { Meta } from "./Meta";
import { Tags } from "./Tags";

import * as styles from "./Post.module.scss";

type Props = {
  post: Node;
};

const Post: React.FC<Props> = ({ post }) => {
  const { html } = post;
  const { tagSlugs, slug } = post.fields;
  const { tags, title, date } = post.frontmatter;

  return (
    <div className={styles.post}>
      <Link className={styles.button} to="/">
        文章列表
      </Link>

      <div className={styles.content}>
        <Content body={html} title={title} />
      </div>

      <div className={styles.footer}>
        <Meta date={date} />
        {tags && tagSlugs && <Tags tags={tags} tagSlugs={tagSlugs} />}
        <Author />
      </div>

      <div className={styles.comments}>
        <Comments postSlug={slug} postTitle={post.frontmatter.title} />
      </div>
    </div>
  );
};

export default Post;
