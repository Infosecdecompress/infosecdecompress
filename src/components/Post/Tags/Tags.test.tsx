import React from "react";
import renderer from "react-test-renderer";

import { Tags } from "@/components/Post/Tags";
import * as mocks from "@/mocks";

describe("Tags", () => {
  it("renders correctly", () => {
    const props = {
      tags: mocks.markdownRemark.frontmatter.tags,
      tagSlugs: mocks.markdownRemark.fields.tagsSlugs,
    };

    const tree = renderer.create(<Tags {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
