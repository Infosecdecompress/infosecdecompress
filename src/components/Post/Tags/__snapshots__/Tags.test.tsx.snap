// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags renders correctly 1`] = `
<div>
  <ul>
    <li>
      <a
        href="/handwriting"
      >
        Handwriting
      </a>
    </li>
    <li>
      <a
        href="/helvetica"
      >
        Helvetica
      </a>
    </li>
  </ul>
</div>
`;
